java/compaction-core/src/main/java/sleeper/compaction/task/CompactionTaskStatus.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2022 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sleeper.compaction.task;
+
+import java.time.Instant;
+import java.util.Objects;
+import java.util.UUID;
+
+public class CompactionTaskStatus {
+    private final String taskId;
+    private final CompactionTaskStartedStatus startedStatus;
+    private final CompactionTaskFinishedStatus finishedStatus;
+    private final Instant expiryDate;
+
+    private CompactionTaskStatus(Builder builder) {
+        taskId = builder.taskId;
+        startedStatus = builder.startedStatus;
+        finishedStatus = builder.finishedStatus;
+        expiryDate = builder.expiryDate;
+    }
+
+    public static CompactionTaskStatus.Builder started(long startTime) {
+        return builder().taskId(UUID.randomUUID().toString()).startedStatus(
```

Comment:
Is there a better identifier we could use for the task so that someone could find the logs for it?

Reply:
It's probably okay since it will still add value to tie the task to the job.